pkg/raft/raft.go

```diff
@@ -398,7 +398,7 @@ type raft struct {
 	// the current leader or if the leader is fortified when ticked.
 	//
 	// Invariant: electionElapsed = 0 when r.leadEpoch != 0 on a follower.
-	electionElapsed int
+	electionElapsed uint64
```

Comment:
We're performing arithmetic on these fields, so I think we should make them `int64`s, not `uint64`s. Unsigned integers are generally frowned upon as a means of encoding "non-negative semantics". See https://graphitemaster.github.io/aau/ and https://www.learncpp.com/cpp-tutorial/unsigned-integers-and-why-to-avoid-them for some discussion why. I recall there was also a similar best practice at Google. They don't actually enforce that values don't go negative, they just underflow and cause such bugs to be harder to catch. It's also common to want to use a negative offset when manipulating values that should not go negative.